client/src/components/App.jsx

```diff
@@ -17,22 +17,23 @@ class App extends React.Component {
   }
 
   componentDidMount() {
-    const url = window.location.href.split("/");
-    const id = Math.floor(Math.random() * 10000);
+    const url = window.location.href.split('/');
+    const id = url[url.length - 1];
     console.log('id --->', id);
     console.log('isLoading state -->', this.state.isLoading);
-    axios
-      .get(`/data`) // <---- change this back to ${id}
+    const instance = axios.create({ baseURL: `ec2-18-223-237-77.us-east-2.compute.amazonaws.com:${port}` });
+    instance
+      .get(`/product/data/${id}`) // <---- change this back to ${id}
```

Comment:
Can probably remove the comment now that you've added ${id} back in.
"All comments are lies"